File: js/org/jmol/exportjs/_JSExporter.js
Clazz.declarePackage ("org.jmol.exportjs");
Clazz.load (["org.jmol.exportjs.__CartesianExporter", "java.util.Hashtable", "org.jmol.exportjs.Export3D", "$.___Exporter"], "org.jmol.exportjs._JSExporter", ["java.lang.Boolean", "$.Float", "org.jmol.exportjs.UseTable", "org.jmol.util.ColorUtil"], function () {
c$ = Clazz.decorateAsClass (function () {
this.useTable = null;
this.htSpheresRendered = null;
this.htObjects = null;
this.ret = null;
Clazz.instantialize (this, arguments);
}, org.jmol.exportjs, "_JSExporter", org.jmol.exportjs.__CartesianExporter);
Clazz.prepareFields (c$, function () {
this.htSpheresRendered =  new java.util.Hashtable ();
this.htObjects =  new java.util.Hashtable ();
this.ret =  new Array (1);
});
Clazz.makeConstructor (c$, 
function () {
Clazz.superConstructor (this, org.jmol.exportjs._JSExporter, []);
this.useTable =  new org.jmol.exportjs.UseTable ();
});
Clazz.overrideMethod (c$, "outputFace", 
function (is, coordMap, faceVertexMax) {
}, "~A,~A,~N");
Clazz.defineMethod (c$, "useSphere", 
($fz = function (id, isNew, pt, o) {
System.out.println (id + " " + isNew + " " + pt + " " + o);
}, $fz.isPrivate = true, $fz), "~S,~B,javax.vecmath.Point3f,~A");
Clazz.defineMethod (c$, "useCylinder", 
($fz = function (id, isNew, pt1, pt2, o) {
System.out.println (id + " " + isNew + " " + pt1 + " " + pt2 + " " + o);
}, $fz.isPrivate = true, $fz), "~S,~B,javax.vecmath.Point3f,javax.vecmath.Point3f,~A");
Clazz.overrideMethod (c$, "outputSphere", 
function (ptCenter, radius, colix, checkRadius) {
var iRad = Math.round ((radius * 100));
var check = org.jmol.exportjs.___Exporter.round (ptCenter) + (checkRadius ? " " + iRad : "");
if (this.htSpheresRendered.get (check) != null) return ;
this.htSpheresRendered.put (check, Boolean.TRUE);
var found = this.useTable.getDef ("S" + colix + "_" + iRad, this.ret);
var o;
if (found) o = this.htObjects.get (this.ret[0]);
 else this.htObjects.put (this.ret[0], o = [this.getColor (colix), Float.$valueOf (radius)]);
this.useSphere (this.ret[0], !found, ptCenter, o);
}, "javax.vecmath.Point3f,~N,~N,~B");
Clazz.overrideMethod (c$, "outputCylinder", 
function (ptCenter, pt1, pt2, colix, endcaps, radius, ptX, ptY, checkRadius) {
if (ptX != null) return false;
var length = pt1.distance (pt2);
var found = this.useTable.getDef ("C" + colix + "_" + Math.round ((length * 100)) + "_" + radius + "_" + endcaps, this.ret);
var o;
if (found) o = this.htObjects.get (this.ret[0]);
 else this.htObjects.put (this.ret[0], o = [this.getColor (colix),  new Float (length),  new Float (radius)]);
this.useCylinder (this.ret[0], !found, pt1, pt2, o);
return true;
}, "javax.vecmath.Point3f,javax.vecmath.Point3f,javax.vecmath.Point3f,~N,~N,~N,javax.vecmath.Point3f,javax.vecmath.Point3f,~B");
Clazz.overrideMethod (c$, "outputCircle", 
function (pt1, pt2, radius, colix, doFill) {
}, "javax.vecmath.Point3f,javax.vecmath.Point3f,~N,~N,~B");
Clazz.overrideMethod (c$, "outputCone", 
function (ptBase, ptTip, radius, colix) {
}, "javax.vecmath.Point3f,javax.vecmath.Point3f,~N,~N");
Clazz.overrideMethod (c$, "outputEllipsoid", 
function (center, points, colix) {
}, "javax.vecmath.Point3f,~A,~N");
Clazz.defineMethod (c$, "getColor", 
($fz = function (colix) {
return org.jmol.util.ColorUtil.colorPointFromInt2 (this.g3d.getColorArgbOrGray (colix));
}, $fz.isPrivate = true, $fz), "~N");
Clazz.defineMethod (c$, "addObject", 
function (id, o) {
{
}}, "~S,~O");
Clazz.overrideMethod (c$, "outputTextPixel", 
function (pt, argb) {
}, "javax.vecmath.Point3f,~N");
Clazz.overrideMethod (c$, "outputTriangle", 
function (pt1, pt2, pt3, colix) {
}, "javax.vecmath.Point3f,javax.vecmath.Point3f,javax.vecmath.Point3f,~N");
Clazz.defineMethod (c$, "output", 
function (pt) {
}, "javax.vecmath.Tuple3f");
Clazz.overrideMethod (c$, "outputHeader", 
function () {
});
});

File: src/org/jmol/exportjs/_JSExporter.java
/* $RCSfile$
 * $Author: hansonr $
 * $Date: 2007-05-18 15:41:42 -0500 (Fri, 18 May 2007) $
 * $Revision: 7752 $

 *
 * Copyright (C) 2003-2005  The Jmol Development Team
 *
 * Contact: jmol-developers@lists.sf.net
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package org.jmol.exportjs;


import javax.vecmath.Point3f;
import javax.vecmath.Tuple3f;

import org.jmol.util.J2SRequireImport;

@J2SRequireImport({org.jmol.exportjs.___Exporter.class, org.jmol.exportjs.__CartesianExporter.class, org.jmol.exportjs.Export3D.class})
public class _JSExporter extends __CartesianExporter {

  public _JSExporter() {
  }

	@Override
	protected void outputCircle(Point3f pt1, Point3f pt2, float radius,
			short colix, boolean doFill) {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected void outputCone(Point3f ptBase, Point3f ptTip, float radius,
			short colix) {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected boolean outputCylinder(Point3f ptCenter, Point3f pt1, Point3f pt2,
			short colix1, byte endcaps, float radius, Point3f ptX, Point3f ptY,
			boolean checkRadius) {
		// TODO Auto-generated method stub
		return false;
	}

	@Override
	protected void outputEllipsoid(Point3f center, Point3f[] points, short colix) {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected void outputFace(int[] is, int[] coordMap, int faceVertexMax) {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected void outputSphere(Point3f ptAtom2, float f, short colix,
			boolean checkRadius) {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected void outputTextPixel(Point3f pt, int argb) {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected void outputTriangle(Point3f pt1, Point3f pt2, Point3f pt3,
			short colix) {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected void output(Tuple3f pt) {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected void outputHeader() {
		// TODO Auto-generated method stub
		
	}
  
}


